frontend/vue/components/CodeExercise/CodeOutput.vue

```diff
@@ -43,19 +45,55 @@ export default defineComponent({
     }, (error: Error) => {
       this.error = error.message
     })
+
+    this.apiTokenPromise = getQiskitUser().then(user => user?.apiToken)
   },
   methods: {
+    needsApiToken (code: string) {
+      return this.apiTokenPromise!.then((apiToken?: string) => {
+        return this.isBackendExecution(code) && !apiToken
+      })
+    },
+    isBackendExecution (code: string) {
+      return code.includes(".get_provider('ibm-q')") ||
```

Comment:
From what we talked I think it's only needed the `code.includes('IBMQ.')`. We can avoid to check `.get_provider`.